ui/actions/nuxeo-move-content-to-coldstorage-button.js

```diff
@@ -0,0 +1,157 @@
+import '../widgets/nuxeo-tooltip.js';
+import './nuxeo-action-button-styles.js';
+
+/**
+`nuxeo-move-content-to-coldstorage-button`
+@group Nuxeo UI
+@element nuxeo-move-content-to-coldstorage-button
+*/
+class MoveToColdStorage extends mixinBehaviors([FiltersBehavior, FormatBehavior], Nuxeo.Element) {
+  static get template() {
+    return html`
+        <style include="nuxeo-action-button-styles nuxeo-styles">
+        </style>
+        <nuxeo-operation id="opMove" op="Document.MoveToColdStorage" input="[[document.uid]]"></nuxeo-operation>
+
+        <dom-if if="[[_canMoveToColdStorage(document)]]">
+          <template>
+            <div class="action" on-click="_toggleDialog" hidden$="[[!_canMoveToColdStorage(document)]]">
+              <paper-icon-button icon="nuxeo:coldstorage" noink></paper-icon-button>
+              <span class="label">[[i18n('documentContent.moveToColdStorage.label')]]</span>
+              <nuxeo-tooltip>[[i18n('documentContent.moveToColdStorage.tooltip')]]</nuxeo-tooltip>
+            </div>
+          </template>
+        </dom-if>
+
+        <nuxeo-dialog id="content_to_move_dialog" with-backdrop>
+          <div>
+            <h2>[[i18n('documentContent.moveToColdStorage.ask.confirmation')]]</h2>
+          </div>
+          <p class="horizontal">
+             [[i18n('documentContent.moveToColdStorage.description')]]
+          </p>
+          </div>
+
+          <div class="buttons">
+           <paper-button id="cancel" name="cancel" noink dialog-dismiss>
+             [[i18n('documentContent.moveToColdStorage.cancel')]]
+            </paper-button>
+           <paper-button id="confirm" name="confirm" class="primary" noink dialog-confirm on-click="_toggle">
+              [[i18n('documentContent.moveToColdStorage.confirm')]]
+           </paper-button>
+          </div>
+        </nuxeo-dialog>
+
+         <nuxeo-dialog id="cannot_move_content_dialog" with-backdrop>
+          <p class="horizontal">
+             [[i18n('documentContent.moveToColdStorage.error.cannotPerformMove')]]
+          </p>
+          </div>
+
+          <div class="buttons">
+           <paper-button id="cancel" name="cancel" noink dialog-dismiss>
+             [[i18n('documentContent.moveToColdStorage.ok')]]
+           </paper-button>
+          </div>
+        </nuxeo-dialog>
+      `;
+  }
+
+  static get is() {
+    return 'nuxeo-move-content-to-coldstorage-button';
+  }
+
+  static get properties() {
+    return {
+      /**
+       * Input document.
+       */
+      document: Object,
+
+      /**
+       * Input user.
+       */
+      user: Object,
+    };
+  }
+
+  _toggleDialog() {
+    this.$.content_to_move_dialog.toggle();
+  }
+
+  _canMoveToColdStorage(document) {
+    return (
+      document &&
+      !this.hasFacet(document, 'ColdStorage') &&
+      (this.hasAdministrationPermissions(this.user) || this.hasPermission(document, 'WriteColdStorage')) &&
+      document.properties['file:content']
```

Comment:
All the filter methods already check `document`, so we may not need this check here.
Also for `document.properties['file:content']` we may want to rely on https://github.com/nuxeo/nuxeo-elements/blob/master/ui/nuxeo-filters-behavior.js#L175-L180